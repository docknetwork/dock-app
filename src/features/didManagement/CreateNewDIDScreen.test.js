import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {
  CreateNewDIDScreen,
  CreateNewDIDScreenContainer,
} from './CreateNewDIDScreen';

const mockStore = configureMockStore();

describe('CreateNewDIDScreen', () => {
  it('should render correctly without modal ', () => {
    const initialState = {
      // placeholder for redux store
    };

    const form = {
      didType: 'didkey',
      showDIDDockQuickInfo: false,
      keypairType: 'ed25519',
      derivationPath: '',
      _errors: {
        didType: '',
      },
      _hasError: false,
    };
    const wrapper = shallow(
      <CreateNewDIDScreen
        handleChange={jest.fn()}
        form={form}
        handleSubmit={jest.fn()}
      />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
  it('should render correctly with modal', () => {
    const initialState = {
      // placeholder for redux store
    };

    const form = {
      didType: 'diddock',
      showDIDDockQuickInfo: true,
      keypairType: 'ed25519',
      derivationPath: '',
      _errors: {
        didType: '',
      },
      _hasError: false,
    };
    const wrapper = shallow(
      <CreateNewDIDScreen
        handleChange={jest.fn()}
        form={form}
        handleSubmit={jest.fn()}
      />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('should render CreateNewDIDScreenContainer correctly', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(<CreateNewDIDScreenContainer />, {
      context: {store: mockStore(initialState)},
    });
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
