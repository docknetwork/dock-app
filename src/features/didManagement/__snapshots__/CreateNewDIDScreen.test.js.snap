// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateNewDIDScreen should render CreateNewDIDScreenContainer correctly 1`] = `
<ScreenContainer
  accessibilityLabel="CreateNewDIDScreen"
  testID="CreateNewDIDScreen"
>
  <Styled(View)>
    <Box
      alignItems="center"
      flexDirection="row"
      marginLeft={1}
      marginRight={22}
    >
      <Memo(ForwardRef(Box))
        width="80px"
      >
        <BackButton
          accessibilityLabel="CreateNewDIDScreenGoBack"
          onPress={[Function]}
          testID="CreateNewDIDScreenGoBack"
        />
      </Memo(ForwardRef(Box))>
      <Memo(ForwardRef(Box))
        alignContent="center"
        alignItems="center"
        flex={1}
        pl={15}
        width="100%"
      >
        <Styled(Component)
          variant="h3"
        >
          Create New DID
        </Styled(Component)>
      </Memo(ForwardRef(Box))>
      <Memo(ForwardRef(Box))
        alignItems="flex-end"
        width="80px"
      />
    </Box>
  </Styled(View)>
  <ForwardRef
    marginLeft={3}
    marginRight={3}
  >
    <Memo(ForwardRef(FormControl))
      isInvalid=""
    >
      <Memo(ForwardRef(Stack))
        mt={7}
      >
        <Memo(ForwardRef(FormControlLabel))>
          DID Type
        </Memo(ForwardRef(FormControlLabel))>
        <Memo(ForwardRef(Select))
          accessibilityLabel="DIDType"
          onValueChange={[Function]}
          selectedValue=""
          testID="DIDType"
        >
          <Memo(ForwardRef(Item))
            label="did:key"
            value="didkey"
          />
          <Memo(ForwardRef(Item))
            label="did:dock"
            value="diddock"
          />
        </Memo(ForwardRef(Select))>
      </Memo(ForwardRef(Stack))>
      <Memo(ForwardRef(FormControlErrorMessage)) />
    </Memo(ForwardRef(FormControl))>
    <DIDAdvancedOptions
      accessibilityLabel="DIDAdvancedOptions"
      form={
        Object {
          "_errors": Object {
            "didType": "",
          },
          "_hasError": false,
          "derivationPath": "",
          "didType": "",
          "keypairType": "ed25519",
          "showDIDDockQuickInfo": true,
        }
      }
      onChange={[Function]}
      testID="DIDAdvancedOptions"
    />
  </ForwardRef>
  <LoadingButton
    accessibilityLabel="CreateNewDIDScreenDIDCreate"
    full={true}
    isDisabled={true}
    mb={70}
    ml={3}
    mr={3}
    onPress={[Function]}
    testID="CreateNewDIDScreenDIDCreate"
  >
    Create
  </LoadingButton>
</ScreenContainer>
`;

exports[`CreateNewDIDScreen should render correctly with modal 1`] = `
<RNCSafeAreaView
  accessibilityLabel="CreateNewDIDScreen"
  style={
    Object {
      "backgroundColor": "#000",
      "flex": 1,
    }
  }
  testID="CreateNewDIDScreen"
>
  <AppGlobalHeader />
  <Memo(ForwardRef(Box))
    flex={1}
  >
    <Memo(ForwardRef(Box))
      flex={1}
    >
      <Styled(View)>
        <Box
          alignItems="center"
          flexDirection="row"
          marginLeft={1}
          marginRight={22}
        >
          <Memo(ForwardRef(Box))
            width="80px"
          >
            <BackButton
              accessibilityLabel="CreateNewDIDScreenGoBack"
              onPress={[Function]}
              testID="CreateNewDIDScreenGoBack"
            />
          </Memo(ForwardRef(Box))>
          <Memo(ForwardRef(Box))
            alignContent="center"
            alignItems="center"
            flex={1}
            pl={15}
            width="100%"
          >
            <Styled(Component)
              variant="h3"
            >
              Create New DID
            </Styled(Component)>
          </Memo(ForwardRef(Box))>
          <Memo(ForwardRef(Box))
            alignItems="flex-end"
            width="80px"
          />
        </Box>
      </Styled(View)>
      <ForwardRef
        marginLeft={3}
        marginRight={3}
      >
        <Memo(ForwardRef(FormControl))
          isInvalid=""
        >
          <Memo(ForwardRef(Stack))
            mt={7}
          >
            <Memo(ForwardRef(FormControlLabel))>
              DID Type
            </Memo(ForwardRef(FormControlLabel))>
            <Memo(ForwardRef(Select))
              accessibilityLabel="DIDType"
              selectedValue="diddock"
              testID="DIDType"
            >
              <Memo(ForwardRef(Item))
                label="did:key"
                value="didkey"
              />
              <Memo(ForwardRef(Item))
                label="did:dock"
                value="diddock"
              />
            </Memo(ForwardRef(Select))>
          </Memo(ForwardRef(Stack))>
          <Memo(ForwardRef(FormControlErrorMessage)) />
        </Memo(ForwardRef(FormControl))>
        <Memo(ForwardRef(VStack))
          mt={7}
          px={5}
          py={5}
          style={
            Object {
              "backgroundColor": "#00174D",
              "borderRadius": 6,
            }
          }
        >
          <Memo(ForwardRef(HStack))>
            <Memo(ForwardRef(HStack))
              style={
                Object {
                  "flexGrow": 1,
                }
              }
            >
              <Memo(ForwardRef(Icon))
                as={[Function]}
                mt={2}
                name="information-circle-outline"
              />
              <Styled(Component)
                ml={1}
                mt={2}
                variant="h3"
              >
                 Quick Info
              </Styled(Component)>
            </Memo(ForwardRef(HStack))>
            <IconButton
              accessibilityLabel="CreateNewDIDScreenCloseQuickInfo"
              onPress={[Function]}
              testID="CreateNewDIDScreenCloseQuickInfo"
            >
              <Memo(ForwardRef(Icon))
                as={[Function]}
                name="close-outline"
              />
            </IconButton>
          </Memo(ForwardRef(HStack))>
          <Styled(Component)
            mt={3}
            textAlign="left"
            variant="didDescription"
          >
            did:dock requires DOCK token to create. This is used for the transaction fee required to create the DID and have a record on the blockchain.
          </Styled(Component)>
          <Styled(Component)
            color="#fff"
            my={2}
          >
            Learn More 
            <Memo(ForwardRef(Icon))
              as={[Function]}
              name="open-outline"
              size="xs"
            />
          </Styled(Component)>
        </Memo(ForwardRef(VStack))>
        <DIDAdvancedOptions
          accessibilityLabel="DIDAdvancedOptions"
          form={
            Object {
              "_errors": Object {
                "didType": "",
              },
              "_hasError": false,
              "derivationPath": "",
              "didType": "diddock",
              "keypairType": "ed25519",
              "showDIDDockQuickInfo": true,
            }
          }
          onChange={
            [MockFunction] {
              "calls": Array [
                Array [
                  "didType",
                ],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            }
          }
          testID="DIDAdvancedOptions"
        />
      </ForwardRef>
      <LoadingButton
        accessibilityLabel="CreateNewDIDScreenDIDCreate"
        full={true}
        isDisabled={false}
        mb={70}
        ml={3}
        mr={3}
        onPress={[MockFunction]}
        testID="CreateNewDIDScreenDIDCreate"
      >
        Create
      </LoadingButton>
    </Memo(ForwardRef(Box))>
  </Memo(ForwardRef(Box))>
</RNCSafeAreaView>
`;

exports[`CreateNewDIDScreen should render correctly without modal  1`] = `
<RNCSafeAreaView
  accessibilityLabel="CreateNewDIDScreen"
  style={
    Object {
      "backgroundColor": "#000",
      "flex": 1,
    }
  }
  testID="CreateNewDIDScreen"
>
  <AppGlobalHeader />
  <Memo(ForwardRef(Box))
    flex={1}
  >
    <Memo(ForwardRef(Box))
      flex={1}
    >
      <Styled(View)>
        <Box
          alignItems="center"
          flexDirection="row"
          marginLeft={1}
          marginRight={22}
        >
          <Memo(ForwardRef(Box))
            width="80px"
          >
            <BackButton
              accessibilityLabel="CreateNewDIDScreenGoBack"
              onPress={[Function]}
              testID="CreateNewDIDScreenGoBack"
            />
          </Memo(ForwardRef(Box))>
          <Memo(ForwardRef(Box))
            alignContent="center"
            alignItems="center"
            flex={1}
            pl={15}
            width="100%"
          >
            <Styled(Component)
              variant="h3"
            >
              Create New DID
            </Styled(Component)>
          </Memo(ForwardRef(Box))>
          <Memo(ForwardRef(Box))
            alignItems="flex-end"
            width="80px"
          />
        </Box>
      </Styled(View)>
      <ForwardRef
        marginLeft={3}
        marginRight={3}
      >
        <Memo(ForwardRef(FormControl))
          isInvalid=""
        >
          <Memo(ForwardRef(Stack))
            mt={7}
          >
            <Memo(ForwardRef(FormControlLabel))>
              DID Type
            </Memo(ForwardRef(FormControlLabel))>
            <Memo(ForwardRef(Select))
              accessibilityLabel="DIDType"
              selectedValue="didkey"
              testID="DIDType"
            >
              <Memo(ForwardRef(Item))
                label="did:key"
                value="didkey"
              />
              <Memo(ForwardRef(Item))
                label="did:dock"
                value="diddock"
              />
            </Memo(ForwardRef(Select))>
          </Memo(ForwardRef(Stack))>
          <Memo(ForwardRef(FormControlErrorMessage)) />
        </Memo(ForwardRef(FormControl))>
        <DIDAdvancedOptions
          accessibilityLabel="DIDAdvancedOptions"
          form={
            Object {
              "_errors": Object {
                "didType": "",
              },
              "_hasError": false,
              "derivationPath": "",
              "didType": "didkey",
              "keypairType": "ed25519",
              "showDIDDockQuickInfo": false,
            }
          }
          onChange={
            [MockFunction] {
              "calls": Array [
                Array [
                  "didType",
                ],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            }
          }
          testID="DIDAdvancedOptions"
        />
      </ForwardRef>
      <LoadingButton
        accessibilityLabel="CreateNewDIDScreenDIDCreate"
        full={true}
        isDisabled={false}
        mb={70}
        ml={3}
        mr={3}
        onPress={[MockFunction]}
        testID="CreateNewDIDScreenDIDCreate"
      >
        Create
      </LoadingButton>
    </Memo(ForwardRef(Box))>
  </Memo(ForwardRef(Box))>
</RNCSafeAreaView>
`;
